sci-misc/boinc/boinc-7.4.42.ebuild

```diff
@@ -0,0 +1,147 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+#WANT_AUTOMAKE="1.11"
+
+AUTOTOOLS_AUTORECONF=true
+
+inherit autotools-utils eutils flag-o-matic systemd user versionator wxwidgets
+
+MY_PV=$(get_version_component_range 1-2)
+
+DESCRIPTION="The Berkeley Open Infrastructure for Network Computing"
+HOMEPAGE="http://boinc.ssl.berkeley.edu/"
+SRC_URI="https://github.com/BOINC/boinc/archive/client_release/${MY_PV}/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="X cuda static-libs systemd"
+
+RDEPEND="
+	!sci-misc/boinc-bin
+	!app-admin/quickswitch
+	>=app-misc/ca-certificates-20080809
+	dev-libs/openssl:*
+	net-misc/curl[ssl,-gnutls(-),-nss(-),curl_ssl_openssl(+)]
+	sys-apps/util-linux
+	sys-libs/zlib
+	cuda? (
+		>=dev-util/nvidia-cuda-toolkit-2.1
+		>=x11-drivers/nvidia-drivers-180.22
+	)
+	X? (
+		dev-db/sqlite:3
+		media-libs/freeglut
+		sys-libs/glibc:2.2
+		virtual/jpeg:*
+		x11-libs/gtk+:2
+		>=x11-libs/libnotify-0.7
+		x11-libs/wxGTK:2.8[X,opengl]
+	)
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+	app-text/docbook-xml-dtd:4.4
+	app-text/docbook2X
+"
+
+S="${WORKDIR}/${PN}-client_release-${MY_PV}-${PV}"
+
+AUTOTOOLS_IN_SOURCE_BUILD=1
+
+src_prepare() {
+	# prevent bad changes in compile flags, bug 286701
+	sed -i -e "s:BOINC_SET_COMPILE_FLAGS::" configure.ac || die "sed failed"
+
+	autotools-utils_src_prepare
+}
+
+src_configure() {
+	local wxconf=""
+
+	# add gtk includes
+	use X && append-flags "$(pkg-config --cflags gtk+-2.0)"
+
+	# look for wxGTK
+	if use X; then
+		WX_GTK_VER="2.8"
+		need-wxwidgets unicode
+		wxconf+=" --with-wx-config=${WX_CONFIG}"
+	else
+		wxconf+=" --without-wxdir"
+	fi
+
+	local myeconfargs=(
+		--disable-server
+		--enable-client
+		--enable-dynamic-client-linkage
+		--disable-static
+		--enable-unicode
+		--with-ssl
+		$(use_with X x)
+		$(use_enable X manager)
+		${wxconf}
+	)
+	autotools-utils_src_configure
+}
+
+src_install() {
+	autotools-utils_src_install
+
+	dodir /var/lib/${PN}/
+	keepdir /var/lib/${PN}/
+
+	if use X; then
+		newicon "${S}"/packages/generic/sea/${PN}mgr.48x48.png ${PN}.png || die
+		make_desktop_entry boincmgr "${PN}" "${PN}" "Math;Science" "Path=/var/lib/${PN}"
+	fi
+
+	# cleanup cruft
+	rm -rf "${ED}"/etc/
+
+	newinitd "${FILESDIR}"/${PN}.init ${PN}
+	newconfd "${FILESDIR}"/${PN}.conf ${PN}
+	use systemd && systemd_dounit "${FILESDIR}"/${PN}.service
+}
+
+pkg_preinst() {
+	enewgroup ${PN}
+	# note this works only for first install so we have to
+	# elog user about the need of being in video group
+	if use cuda; then
+		enewuser ${PN} -1 -1 /var/lib/${PN} "${PN},video"
+	else
+		enewuser ${PN} -1 -1 /var/lib/${PN} "${PN}"
+	fi
+}
+
+pkg_postinst() {
+	echo
+	elog "You are using the source compiled version of ${PN}."
```

Comment:
It'd be better to replace most of ${PN} with verbose boinc. It will make it easier to read this and to check line width.